lib/web/app/auth.go

```diff
@@ -229,15 +229,20 @@ func (h *Handler) completeAppAuthExchange(w http.ResponseWriter, r *http.Request
 
 		webLauncherURLParams := launcherURLParams{
 			publicAddr:          nextRequiredApp,
-			clusterName:         h.clusterName,
 			requiredAppFQDNs:    strings.Join(requiredApps, ","),
 			requiresAppRedirect: true,
 		}
 		addr, err := utils.ParseAddr(webLauncherURLParams.publicAddr)
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		urlString := makeAppRedirectURL(r, h.c.WebPublicAddr, addr.Host(), webLauncherURLParams)
+
+		var proxyPublicAddrs []string
+		for _, addr := range h.c.ProxyPublicAddrs {
```

Comment:
nit: `addr` is locally scoped here but still good to avoid shadowing same name that is used in 235
```suggestion
		for _, pubAddr := range h.c.ProxyPublicAddrs {
```